activerecord/lib/active_record/attribute.rb

```diff
@@ -111,7 +111,10 @@ def hash
     protected
 
     attr_reader :original_attribute
-    alias_method :assigned?, :original_attribute
+
+    def assigned?
+      !original_attribute.nil?
```

Comment:
I agree that this method should not have needed to change. `accessed_fields` isn't documented as returning a boolean, and I don't want to change it to be.